docusaurus/docs/create-a-plugin/extend-a-plugin/fetch-data-from-frontend.md

```diff
@@ -0,0 +1,287 @@
+---
+id: fetch-data-from-frontend
+title: Fetching data from frontend code
+description: Learn how to use the data proxy API to fetch data from frontend code in data source and app plugins in Grafana and overcome CORS limitations
+keywords:
+  - grafana
+  - plugins
+  - data proxy
+  - frontend
+  - data source
+  - CORS
+---
+
+# Fetching data from frontend data source and app plugins
+
+To fetch data from your data source an app plugins you have two options:
+
+- Use the browser [Fetch API](https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API) as you would do for any other javascript project
+- Use the Grafana data proxy.
+
+This page will guide you through how to use the data proxy to fetch data.
+
+## Why use the data proxy?
+
+If you have a data source or app plugin with frontend code and you need to fetch data, you might have encountered cross-site (CORS) limitations or the need to perform authenticated requests.
+
+For these use cases, you want to make use of the data proxy API.
+
+## How does it work?
+
+Instead of performing a request directly from the browser to the server, you perform the request through the Grafana backend server, which will handle it and return the response to the plugin.
+
+### Without data proxy
+
+```mermaid
```

Comment:
Do we support mermaid diagrams in our doc website?

Reply:
Run the docs locally and you'll notice I guess

Reply:
It does not :( I guess I'll remove the diagrams.